panel/lxqtpanel.cpp

```diff
@@ -282,6 +282,13 @@ LXQtPanel::LXQtPanel(const QString &configGroup, LXQt::Settings *settings, QWidg
                 mShowDelayTimer.start();
         }
     });
+    connect(KWindowSystem::self(), &KWindowSystem::showingDesktopChanged, this, [this] (bool showing) {
+        if (showing && mHidable && mHideOnOverlap) {
+            mHideTimer.stop();
+            if (mHidden)
+                mShowDelayTimer.start();
+        }
+    });
```

Comment:
This code may not be compatible with all WMs. It may show the panel on showing desktop but there's no guarantee that the panel is made hidden again on toggling the desktop state *when there is an overlap*. A WM can show desktop without changing `NET::WMGeometry` or `NET::WMState`, as KWin does.

Reply:
> This code may not be compatible with all WMs. 
i tested on kwin, openbox and xfwm4, all of them were showing (actually it was showing panel even without the patch, but didn't when i was reproducing the issue reported), but kwin would autohide again, added a new commit to prevent that.

Reply:
Panel is overlapped (and so, hidden) → Desktop is shown → Panel is shown (sadly, by force). So far, so good. But, with this code, there's no guarantee that Panel is hidden again after Desktop is hidden.

Reply:
What Compiz does without needing a patch:

https://user-images.githubusercontent.com/981076/146638951-2cb937c7-fdeb-4864-9ee7-d49d0bcfe553.mp4

Compiz removes overlapping and restores it. Another WM may neither remove nor restore it.